src/main/docker/readJenkinsVersion.groovy

```diff
@@ -0,0 +1,31 @@
+import java.util.zip.*
+import java.util.jar.*
+
+static String readManifest(String sourceJARFile) throws IOException {
+    def zipFile = new ZipFile(sourceJARFile)
```

Comment:
Sigh, no `try`-with-resources in Groovy? There must be some `with` syntax or something for `AutoCloseable`.

Reply:
Yes, there is a replacement in Groovy. I just decided to follow the Java6 style so that it's simple for the most of Jenkins contributors